build(deps): remove dokka, no longer in use

mikehardy opened the pull request:
<!--- Please fill the necessary details below -->
## Purpose / Description

It doesn't appear that we actually *use* dokka anymore, remove the dependency

## Fixes

- Closes #17612 

## Approach

Remove all references to dokka in our build files

## How Has This Been Tested?

We use jitpack.io to publish the API artifacts, and in our `jitpack.yml` we run `./gradlew :api:publishToMavenLocal`

https://github.com/ankidroid/Anki-Android/blob/ff607b3f4411e47f403f671861435ab29eca4327/jitpack.yml#L15

So I ran that then checked the javadoc artifact (attached) - it does generate (which is job number one) and it appears fine.

To be clear though, I'm not that interested in the output, it's mainly generated at all because some publishers (maven at least, jitpack unknown) require you to have a javadoc jar.

But we also publish the *actual sources* of the API module since we are open source, and most (all?) IDEs will auto-pull those and cross-link everything up and let you just open things and look. Which is 100 times more useful than javadoc and the primary reason why I haven't actually looked at anyone's javadoc in ages.

[api-2.0.0-javadoc.zip](https://github.com/user-attachments/files/18183642/api-2.0.0-javadoc.zip)

## Learning (optional, can help others)

I'd almost forgotten that javadoc existed, but then I don't work much with closed-source products.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

david-allison reviewed (changes requested):
I approve https://github.com/ankidroid/Anki-Android/pull/17621/commits/3808c7c7bff3226b4c6ed5fd0e0436e646964e14, but it feels like this PR needs a rebase before it goes in

mikehardy commented:
yeah the other dependency-updates traffic blew this up

I just squashed dependency-updates to main
then I force-pushed dependency-updates after a reset --hard from main (so they were in sync post-squash)
then I took this branch and dropped all the single-commit dependency-updates before rebasing it from the newly-pushed dependency-updates with the squash

then I pushed that, so should be sparkly clean
